server/routes/user_router.go

```diff
@@ -0,0 +1,229 @@
+package routes
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/rawdaGastan/grid3_auto_deployer/models"
+)
+
+type SignUpInput struct {
+	Name            string `json:"name" binding:"required"`
+	Email           string `json:"email" gorm:"unique" binding:"required"`
+	Password        string `json:"password" binding:"required"`
+	ConfirmPassword string `json:"confirmPassword" binding:"required"`
+}
+
+type AuthData struct {
+	Email string `json:"email"`
+	Code  int    `json:"code"`
+}
+
+type ChangePassword struct {
+	Email           string `json:"email"`
+	Password        string `json:"password" binding:"required"`
+	ConfirmPassword string `json:"confirmPassword" binding:"required"`
+}
+
+func (router *Router) SignUpHandler(w http.ResponseWriter, r *http.Request) {
+	signUp := SignUpInput{}
+	err := json.NewDecoder(r.Body).Decode(&signUp)
+	if err != nil {
+		errJSON, _ := json.Marshal(ErrorMsg{Message: err.Error()})
```

Comment:
we can create an errorResponse function to be used for all error reponses, for example

```
func writeErrRes(w http.ResponseWriter, err error) {
	jsonErrRes, _ := json.Marshal(ErrRes{err.Error()})
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusInternalServerError)
	w.Write(jsonErrRes)
}
```